### PR TITLE
Platform docs cleanup

### PR DESCRIPTION
**Pull request by theotherjimmy**
- Break platform into 3 major sections
 - Introcude relevent concetps in those sections
 - Move detailed exampes into Callback docs

### CODE REVIEW

**Comment by theotherjimmy** [comment]
@AnotherButler I think I got to all of the cleanup I needed to.

**Comment by theotherjimmy** [comment]
@sg- Could you take a look?

File: docs/reference/api/platform/Callback.md [hunk 1]
```diff
@@ -27,3 +27,156 @@ Here is an example that uses everything discussed in the [introduction to callba
 
 - [Platform overview](/docs/development/reference/platform.html).
```
Comment (sg-) [inline docs/reference/api/platform/Callback.md]: This and Intro + related content should now be deleted

File: docs/reference/api/platform/Callback.md [hunk 2]
```diff
@@ -27,3 +27,156 @@ Here is an example that uses everything discussed in the [introduction to callba
...
+Callback<int(float)> cb;          // A callback that takes in a float and returns an int
+Callback<void(float)> cb;         // A callback that takes in a float and returns nothing
+Callback<int()> cb;               // A callback that takes in nothing and returns an int
+Callback<void(float, float)> cb;  // A callback that takes in two floats and returns nothing
+```
+
+You can create a Callback directly from a C function or function pointer with the same type:
+
+``` c++
+void dosomething(int) {
```
Comment (sg-) [inline docs/reference/api/platform/Callback.md]: Great to move the content around but this still violates the no foo/bar rule imo. Make sure to keep another task to make the example useful or tangible

File: docs/reference/api/platform/Callback.md (hunk 1 above)
Comment (theotherjimmy) [reply docs/reference/api/platform/Callback.md]: Thanks! I'll do that.

File: docs/reference/api/platform/Callback.md (hunk 2 above)
Comment (theotherjimmy) [reply docs/reference/api/platform/Callback.md]: I think I'd rather just do that now.

**Pull request merged**